protocols/imap/src/main/java/org/apache/james/imap/decode/parser/ImapParserFactory.java

```diff
@@ -36,77 +39,81 @@
-        imapCommands.put(ImapConstants.SEARCH_COMMAND_NAME, new SearchCommandParser(statusResponseFactory));
-        imapCommands.put(ImapConstants.FETCH_COMMAND_NAME, new FetchCommandParser(statusResponseFactory));
-        imapCommands.put(ImapConstants.STORE_COMMAND_NAME, new StoreCommandParser(statusResponseFactory));
-        imapCommands.put(ImapConstants.UID_COMMAND_NAME, new UidCommandParser(this, statusResponseFactory));
-        imapCommands.put(ImapConstants.IDLE_COMMAND_NAME, new IdleCommandParser(statusResponseFactory));
-        imapCommands.put(ImapConstants.STARTTLS, new StartTLSCommandParser(statusResponseFactory));
-
-        // RFC3691
-        imapCommands.put(ImapConstants.UNSELECT_COMMAND_NAME, new UnselectCommandParser(statusResponseFactory));
-
-        // RFC4978
-        imapCommands.put(ImapConstants.COMPRESS_COMMAND_NAME, new CompressCommandParser(statusResponseFactory));
-
-        imapCommands.put(ImapConstants.ENABLE_COMMAND_NAME, new EnableCommandParser(statusResponseFactory));
-
-        // RFC2087
-        // GETQUOTAROOT, GETQUOTA, SETQUOTA
-        imapCommands.put(ImapConstants.GETQUOTAROOT_COMMAND_NAME, new GetQuotaRootCommandParser(statusResponseFactory));
-        imapCommands.put(ImapConstants.GETQUOTA_COMMAND_NAME, new GetQuotaCommandParser(statusResponseFactory));
-        imapCommands.put(ImapConstants.SETQUOTA_COMMAND_NAME, new SetQuotaCommandParser(statusResponseFactory));
-
-        //RFC5464
-        //SETMETADATA, GETMETADATA
-        imapCommands.put(ImapConstants.SETANNOTATION_COMMAND_NAME, new SetAnnotationCommandParser(statusResponseFactory));
-        imapCommands.put(ImapConstants.GETANNOTATION_COMMAND_NAME, new GetAnnotationCommandParser(statusResponseFactory));
+        ImmutableList<AbstractImapCommandParser> parsers = ImmutableList.of(
+            // Commands valid in any state
+            // CAPABILITY, NOOP, and LOGOUT
+            new CapabilityCommandParser(statusResponseFactory),
+            new NoopCommandParser(statusResponseFactory),
+            new LogoutCommandParser(statusResponseFactory),
+
+            // Commands valid in NON_AUTHENTICATED state.
+            // AUTHENTICATE and LOGIN
+            new AuthenticateCommandParser(statusResponseFactory),
+            new LoginCommandParser(statusResponseFactory),
+
+            // Commands valid in AUTHENTICATED or SELECTED state.
+            // RFC2060: SELECT, EXAMINE, CREATE, DELETE, RENAME, SUBSCRIBE,
+            // UNSUBSCRIBE, LIST, LSUB, STATUS, and APPEND
+            new SelectCommandParser(statusResponseFactory),
+            new ExamineCommandParser(statusResponseFactory),
+            new CreateCommandParser(statusResponseFactory),
+            new DeleteCommandParser(statusResponseFactory),
+            new RenameCommandParser(statusResponseFactory),
+            new SubscribeCommandParser(statusResponseFactory),
+            new UnsubscribeCommandParser(statusResponseFactory),
```

Comment:
oh, it's miraculously back :)

Reply:
I miraculously ran the MPT tests!

Reply:
:godmode:

Reply:
I am so ashamed that I fixed the history....